src/tcompiler.cpp

```diff
@@ -3186,10 +3190,10 @@ static int terra_disassemble(lua_State *L) {
 static bool FindLinker(terra_State *T, LLVM_PATH_TYPE *linker, const char *target) {
 #ifndef _WIN32
 #if LLVM_VERSION >= 36
-    *linker = *sys::findProgramByName("gcc");
+    *linker = *sys::findProgramByName(getenv("CXX"));
```

Comment:
1. Should this be `CC`? I don't think we actually need to require a C++ compiler at runtime.
2. Either way, we should have a fallback in case `CC`/`CXX` is not defined.

Reply:
This logic is essentially the same [as what CMake does](https://github.com/Kitware/CMake/blob/995be1097c35fd3187d147494156bbfb9da7037f/Modules/CMakeDetermineCXXCompiler.cmake). We can switch it to CC and then copy cmake's list of known compilers.

Reply:
I'd be ok with checking `CXX` if `CC` is unset. At that point, I think if the user has a system unusual enough that falling back to `cc` doesn't work, I'm fine with telling them that they have to set one of those two variables.

Reply:
Okay, 192ca54e43eb543fbb6a67c76e1b23432444350f adds a failover here, checking `CC`, then `CXX`, then giving up and demanding `gcc`.

Reply:
Looks like line 3077 doesn't actually check `linker_name`?

Reply:
Fixed